packages/flutter/lib/src/widgets/tap_region.dart

```diff
@@ -329,6 +330,9 @@ class _DummyTapRecognizer extends GestureArenaMember {
 /// regions in the group will act as one.
 ///
 /// If there is no [TapRegionSurface] ancestor, [TapRegion] will do nothing.
+///
+/// [TapRegion] calls [ModalRoute.of] to establish a dependency on the current route,
+/// so that [onTapOutside] isn't called after the user navigates to a different page.
```

Comment:
```suggestion
/// [TapRegion] is aware of the [Route]s in the [Navigator], so that [onTapOutside]
/// isn't called after the user navigates to a different page.
```